src/reference/asciidoc/zookeeper.adoc

```diff
@@ -0,0 +1,57 @@
+[[zookeeper]]
+== Zookeeper Support
+
+=== Introduction
+
+https://zookeeper.apache.org/[Zookeeper] support was added to the framework in _version 4.2_, comprised of:
+
+* MetadataStore
+* LockRegistry
+* Leadership Event Handling
+
+[[zk-metadata-store]]
+=== Zookeeper Metadata Store
+
+The `ZookeeperMetadataStore` can be used where any `MetadataStore` is needed, such as peristent file list filters,
+etc.
+See <<metadata-store>> for more information.
+
+[[zk-lock-registry]]
+=== Zookeeper Lock Registry
+
+The `ZookeeperLockRegistry` can be used where any `LockRegistry` is needed, such as when using an `Aggregator` in a
+clustered environment, with a shared `MessageStore`.
+
+=== Zookeeper Leadership Event Handling
+
+Groups of endpoints can be started/stopped based on leadership being granted or revoked respectively.
+This is useful in clustered scenarios where shared resources must only be consumed by a single instance.
+An example of this is a file inbound channel adapter that is polling a shared directory.
+
+[source, xml]
+----
+<int-zk:leader-listener client="client" path="/siNamespace" role="cluster" />
+----
+
+`client` is a reference to a `CuratorFramework` bean.
+When a leader is elected, an `OnGrantedEvent` will be published for the role `cluster`; any endpoints in that role
+will be started.
+When leadership is revoked, an `OnRevokedEvent` will be published for the role `cluster`; any endpoints in that role
+will be stopped.
+See <<endpoint-roles>> for more information.
+
+The parser for this namespace element creates 3 objects. The equivalent Java configuration would be:
+
+[source, java]
+----
+@Bean
+public LeaderEventPublisher leaderEventPublisher() {
+    return new DefaultLeaderEventPublisher();
+}
+
+@Bean
+public LeaderInitiator leaderInitiator(CuratorFramework client, LeaderEventPublisher eventPublisher) {
+    Candidate candidate = new DefaultCandidate(UUID.randomeUUID().toString(), "cluster"); // unique id, role
+    LeaderInitiator initiator = new LeaderInitiator(client, candidate, "/siNamespace");
```

Comment:
Yes, why not there, if we still have a time a room before their release.
With who I should get in touch to go ahead with request?
Or GH issue is enough?